providers/flagd/pkg/cache/in_memory.go

```diff
@@ -0,0 +1,50 @@
+package cache
+
+import (
+	"context"
+	"sync"
+)
+
+type InMemory struct {
+	values map[string]interface{}
+	rwMux  *sync.RWMutex
```

Comment:
1.19 brings [sync.Map](https://pkg.go.dev/sync#Map) , but I don't know if we want to upgrade just for this.

Reply:
I did consider it initially but
> The Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys. In these two cases, use of a Map may significantly reduce lock contention compared to a Go map paired with a separate Mutex or RWMutex.

convinced me that it wasn't the right use case.
Either way I've since updated this to use generics, will push up soon when I finish the tests 😄

Reply:
> Either way I've since updated this to use generics

Nice! I think I went through the same thing with the flagd-js provider, as I added caching, etc, It became necessary in order to avoid too much duplication.